src/lib/runtime_config/runtime_config.ml

```diff
@@ -1572,134 +1333,242 @@ let make_fork_config ~staged_ledger ~global_slot_since_genesis ~state_hash
-      , "delta": 0
-      , "genesis_state_timestamp": "2019-01-30 12:00:00-08:00" }
-  , "proof":
-      { "level": "check"
-      , "sub_windows_per_window": 8
-      , "ledger_depth": 30
-      , "work_delay": 1
-      , "block_window_duration_ms": 10000
-      , "transaction_capacity": {"2_to_the": 2}
-      , "coinbase_amount": "20"
-      , "supercharged_coinbase_factor": 2
-      , "account_creation_fee": "1" }
-  , "ledger":
-      { "name": "test_split_two_stakers"
-      , "add_genesis_winner": false } }
-      |json}
-      |> Yojson.Safe.from_string |> of_yojson |> Result.ok_or_failwith )
-
-  let delegation =
-    lazy
-      ( (* test_postake_delegation *)
-        {json|
-  { "daemon":
-      { "txpool_max_size": 3000 }
-  , "genesis":
-      { "k": 4
-      , "delta": 0
-      , "slots_per_epoch": 72
-      , "genesis_state_timestamp": "2019-01-30 12:00:00-08:00" }
-  , "proof":
-      { "level": "check"
-      , "sub_windows_per_window": 4
-      , "ledger_depth": 6
-      , "work_delay": 1
-      , "block_window_duration_ms": 5000
-      , "transaction_capacity": {"2_to_the": 2}
-      , "coinbase_amount": "20"
-      , "supercharged_coinbase_factor": 2
-      , "account_creation_fee": "1" }
-  , "ledger":
-      { "name": "test_delegation"
-      , "add_genesis_winner": false } }
-      |json}
-      |> Yojson.Safe.from_string |> of_yojson |> Result.ok_or_failwith )
+  val of_string : string -> t
+
+  val mainnet : t
+
+  val dev : t
+
+  val devnet : t
+
+  val lightnet : t
+end = struct
+  type t =
+    { genesis_constants : Genesis_constants.Inputs.t
+    ; constraint_constants : Genesis_constants.Constraint_constants.Inputs.t
+    }
+
+  let mainnet =
+    { genesis_constants =
+        { genesis_state_timestamp = "2024-06-05T00 =00 =00Z"
+        ; k = 290
+        ; slots_per_epoch = 7140
+        ; slots_per_sub_window = 7
+        ; grace_period_slots = 2160
+        ; delta = 0
+        ; pool_max_size = 3000
+        ; num_accounts = None
+        ; zkapp_proof_update_cost = 10.26
+        ; zkapp_signed_single_update_cost = 9.140000000000001
+        ; zkapp_signed_pair_update_cost = 10.08
+        ; zkapp_transaction_cost_limit = 69.45
+        ; max_event_elements = 100
+        ; max_action_elements = 100
+        ; zkapp_cmd_limit_hardcap = 128
+        ; minimum_user_command_fee = "0.001"
+        }
+    ; constraint_constants =
+        { scan_state_with_tps_goal = true
+        ; scan_state_tps_goal_x10 = Some 2
+        ; block_window_duration = 180000
+        ; scan_state_transaction_capacity_log_2 = None
+        ; supercharged_coinbase_factor = 1
+        ; scan_state_work_delay = 2
+        ; coinbase = "720"
+        ; account_creation_fee_int = "1.0"
+        ; ledger_depth = 35
+        ; sub_windows_per_window = 11
+        ; fork = None
+        ; proof_level = "full"
+        }
+    }
+
+  let devnet =
+    { genesis_constants =
+        { genesis_state_timestamp = "2021-09-24T00=00=00Z"
```

Comment:
Doesn't look right: for mainnet we have current mainnet's timestamp from 2024, for devnet we use the old devnet's value (it was also forked in 2024).

Reply:
I understand it's likely an artifact of `.mlh` files we had, still why not synchronize it with values from `genesis_ledgers/xxx.json`?

Reply:
A suggestion: define `devnet` as `mainnet` with the genesis timestamp override. And then similarly for `dev` and `lightnet` (with more overrides).